woocommerce/woocommerce-ios
# Add site ID to `SiteVisitStats` and `TopEarnerStats`: Networking/Yosemite/app layer changes

Fixes #2916 

Depends on https://github.com/woocommerce/woocommerce-ios/pull/3510

## Why

This PR follows storage layer changes in https://github.com/woocommerce/woocommerce-ios/pull/3510. Since we don't delete all data after switching to another store, adding a site ID to data models associated with a store would prevent issues from incorrect data from another store ([example](https://github.com/woocommerce/woocommerce-ios/issues/2916#issuecomment-765296344)).

## Changes

- Added site ID to Networking models `SiteVisitStats` and `TopEarnerStats`, with corresponding changes in Networking and Yosemite layers
- Added site ID predicate in `TopPerformerDataViewController` in the app layer

## Testing

- Launch the app --> the visitor stats and top performers should look the same as before
- Pull down to refresh dashboard --> the visitor stats and top performers should look the same as before

Update release notes:
